x-pack/test/api_integration/apis/uptime/rest/add_monitor.ts

```diff
@@ -68,6 +69,71 @@ export default function ({ getService }: FtrProviderContext) {
       expect(apiResponse.body.message).eql('Monitor type is invalid');
     });
 
+    it('can create valid monitors without all defaults', async () => {
+      // Delete a required property to make payload invalid
+      const newMonitor = {
+        name: 'Sample name',
+        type: 'http',
+        url: 'https://elastic.co',
+        locations: [
+          {
+            id: 'eu-west-01',
+            label: 'Europe West',
+            geo: {
+              lat: 33.2343132435,
+              lon: 73.2342343434,
+            },
+            url: 'https://example-url.com',
```

Comment:
We don't have a `url` field on http monitor type, it's `urls`.  The result is both `url` field (the provided one) and `urls` (from defaults) get into attributes.

![Screenshot 2022-06-15 at 14 56 44](https://user-images.githubusercontent.com/2748376/173833296-b3f7ae89-35c8-4544-a224-2883c552aa71.png)

The validation passes because io-ts only validates the type and `""` is a valid value for `urls`. We may need to validate the emptiness of required fields after io-ts has performed its validation.

Reply:
Yes, I agree. I think that's out of scope for this ticket though. My thought is that this change is primarily intended to support our internal users creating monitors, and there's less of an expectation of an internal api being fully supported for external use. For this reason, I'm okay with the validation limitations for the internal customer use case. In terms of validation for emptiness, the UI currently does that in our fully-supported use case.

However, I do think we should go back and enhance the validation for the reason you've described.